File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/payment/BuildGetPaymentSourcesRequestTest.java
package org.hyperledger.indy.sdk.payment;

import org.hyperledger.indy.sdk.payments.IncompatiblePaymentException;
import org.hyperledger.indy.sdk.payments.Payments;
import org.hyperledger.indy.sdk.payments.UnknownPaymentMethodException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;

public class BuildGetPaymentSourcesRequestTest extends PaymentIntegrationTest {

	@Test
	public void testBuildGetPaymentSourcesRequestWorksForUnknownPaymentMethod() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(UnknownPaymentMethodException.class));

		Payments.buildGetPaymentSourcesRequest(wallet, DID_TRUSTEE, paymentAddress).get();
	}

	@Test
	public void testBuildGetPaymentSourcesRequestWorksForInvalidPaymentAddress() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(IncompatiblePaymentException.class));

		Payments.buildGetPaymentSourcesRequest(wallet, DID_TRUSTEE, "pay:null1").get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/PaymentTests/BuildGetPaymentSourcesRequestTest.cs
using Hyperledger.Indy.PaymentsApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.PairwiseTests
{
    [TestClass]
    public class BuildGetPaymentSourcesRequestTest : PaymentIntegrationTestBase
    {
        [TestMethod]
        public async Task TestBuildGetPaymentSourcesRequestWorksForUnknownPaymentMethod()
        {
            var ex = await Assert.ThrowsExceptionAsync<UnknownPaymentMethodException>(() =>
                 Payments.BuildGetPaymentSourcesAsync(wallet, DID_TRUSTEE, paymentAddress)
             );
        }

        [TestMethod]
        public async Task TestBuildGetPaymentSourcesRequestWorksForInvalidPaymentAddress()
        {
            var ex = await Assert.ThrowsExceptionAsync<IncompatiblePaymentMethodsException>(() =>
                Payments.BuildGetPaymentSourcesAsync(wallet, DID_TRUSTEE, "pay:null1")
            );
        }
    }
}
